File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <unordered_map>
#include <iostream>
using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    unordered_map <string,int> p;
    unordered_map <string,int> c;
    for (string par:participant){
        p[par]+=1;
    }
    for (string com:completion){
        c[com]+=1;
    }
    for(string par:participant){
        if (p[par]!=c[par]){
            return par;
        }
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";

        HashMap<String, Integer> map = new HashMap<>();

        for (int i = 0; i < participant.length; i++) {
            if(map.containsKey(participant[i])){
                map.put(participant[i],map.get(participant[i]) + 1);
            }
            else{
                map.put(participant[i],1);
            }
        }
        
        for(int j=0; j< completion.length;j++){
            if(map.containsKey(completion[j])){
                map.put(completion[j],map.get(completion[j])-1);
            }
            if(map.get(completion[j])==0){
                map.remove(completion[j]);
            }
        }
        
        for (String key : map.keySet()) {
        return key;
        }

    return "";
        
    }
}